Add function to allow looking up app based on name and owner username

CameronDiver opened the pull request:
This can be useful for looking up shared applications when the user in
question has an application of the same name as the one that is shared.

Connects-to: #349
Change-type: minor
Signed-off-by: Cameron Diver <cameron@resin.io>
---- Autogenerated Waffleboard Connection: Connects to #349

pimterry commented on lib/models/application.coffee:
```diff
@@ -149,6 +149,53 @@ getApplicationModel = (deps, opts) ->
+	exports.getAppWithOwner = (appName, owner, options = {}, callback) ->
+		callback = findCallback(arguments)
+
+		a.pine.get
```
What is `a`? I'm surprised this code works...

CameronDiver replied:
Oh shoot. I've copied the debugging code out of another file. It's essentially the same, but with different vars... Sorry for the noise.

pimterry commented on lib/models/application.coffee:
```diff
@@ -149,6 +149,53 @@ getApplicationModel = (deps, opts) ->
+							$filter:
+								$eq: [
+									$tolower: $: 'username'
+									owner
```
We should `toLowerCase` both the owner and appName here, if we're going to compare them to `$tolower`.

CameronDiver replied:
Same as above, I've copied the wrong code...

CameronDiver commented:
@pimterry I was using resin-sdk-preconfigured to do the testing and accidentally committed the wrong code! Sorry about that. Anyway I've added the correct code and added some tests for the new method.

emirotin commented on lib/models/application.coffee:
```diff
@@ -149,6 +149,56 @@ getApplicationModel = (deps, opts) ->
+	#
+	# @example
+	# resin.models.application.getAppWithOwner('MyApp', 'MyUser').then(function(application) {
+	#		console.log(application);
```
Unneeded double tab?

emirotin commented on tests/integration/models/application.spec.coffee:
```diff
@@ -15,6 +15,13 @@ describe 'Application Model', ->
+				promise = resin.models.application.getAppWithOwner('testapp', 'FooBar')
+				m.chai.expect(promise).to.be.rejected
+
+
```
Extra empty line
